### PR TITLE
parking

### PR DESCRIPTION
**Pull request by Girrajgg**
assignment

### CODE REVIEW

File: parking/src/dao/ChargesDao.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		  		perHourChargesByVehicleTypes.put(VehicleType.VehicleTypes.fourWheeler, 20);
+		  		perHourChargesByVehicleTypes.put(VehicleType.VehicleTypes.sixWheeler, 20);	
+	        }
+	public Integer getChargesByVehilcTypes(VehicleType.VehicleTypes vehicleTypes) throws Exception{
+		Integer charges = perHourChargesByVehicleTypes.get(vehicleTypes);
+		
+		if(charges == null){
+			throw new Exception("Invaild vehicle type");
+		}
+		else
```
Comment (MohitGarg2905) [inline parking/src/dao/ChargesDao.java:25]: else statement not needed, you can directly return charges.

File: parking/src/models/Bike.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+public class Bike extends TwoWheelers {  
+	String regNumber;
+	
+	public Bike(String regNumber){
+		super(regNumber);
+		regNumber = getRegnumber();  	 
+	}
+	
+	void drive(){
```
Comment (MohitGarg2905) [inline parking/src/models/Bike.java:13]: access modifier missing

File: parking/src/models/Floor.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.*;
+import constant.VehicleType;
+import models.ParkingLot;
+
+
+public class Floor {
+	
+	        
+	
+	public static void floorNumber(Vehicle vehicle){
```
Comment (MohitGarg2905) [inline parking/src/models/Floor.java:12]: Wrong method, methods inside models should only set/return the properties declared within them.
There should be a property floorNumber, and the getter methods.

File: parking/src/models/ParkingLot.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	protected static Map<Integer, Integer> slotStatusFloor2 = new HashMap<Integer, Integer>();
+	protected static Map<Integer, Integer> slotStatusFloor3 = new HashMap<Integer, Integer>();
+	protected static Map<Integer, Integer> slotStatusFloor4 = new HashMap<Integer, Integer>();
+	protected static Map<Integer, Integer> slotStatusFloor5 = new HashMap<Integer, Integer>();
+	protected static Map<Integer, Integer> slotStatusFloor6 = new HashMap<Integer, Integer>();
+	protected static Map<Integer, Integer> FloorStatus = new HashMap<Integer, Integer>();
+	
+	
+	//creating slots for each floor
+	public static void makeParkingLot(){
```
Comment (MohitGarg2905) [inline parking/src/models/ParkingLot.java:25]: should not be static, models should not have static methods.

File: parking/src/models/ParkingLot.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+						    for(int i=0; i<10;i++){
+									    	//slotkeyForEachFloor.add(i, 0);
+									    	slotStatusFloor0.put(i, 0);
+									    	slotStatusFloor1.put(i, 0);
+									    	slotStatusFloor2.put(i, 0);
+									    	slotStatusFloor3.put(i, 0);
+									    	slotStatusFloor4.put(i, 0);
+									    	slotStatusFloor5.put(i, 0);
+									    	slotStatusFloor6.put(i, 0);
+									    	FloorStatus.put(i, 0);
```
Comment (MohitGarg2905) [inline parking/src/models/ParkingLot.java:53]: what is this variable for?

File: parking/src/models/Slot.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
+package models;
+
+import java.util.*;
+import models.ParkingLot;
+import models.Vehicle;
+import constant.VehicleType;
+import models.Time;
+
+public class Slot {
+	public static Map<String, Integer> FloorCorrespondingToslotId = new HashMap<String, Integer>();
```
Comment (MohitGarg2905) [inline parking/src/models/Slot.java:10]: variables name should start with small case

File: parking/src/models/Slot.java [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+	String slotId = UUID.randomUUID().toString();
+	    
+ 	//try to give slot
+   	 boolean checkSlotOccupied = ParkingLot.occupied.contains(slotId); 
+        if (checkSlotOccupied)
+        {
+       	 System.out.println("slotoccupied");
+        }
+        else
+        {   	 if(vType == VehicleType.VehicleTypes.twoWheeler)
```
Comment (MohitGarg2905) [inline parking/src/models/Slot.java:28]: if else conditions should not be used here, use proper data structures and pull data directly from them.

File: parking/src/parking/Floor.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package parking;
+
+public class Floor {
```
Comment (MohitGarg2905) [inline parking/src/parking/Floor.java:3]: why an empty class?

File: parking/src/parking/slot.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
+package parking;
+
+import vehicles.Vehicle;
+import java.util.*;
+
+public class slot {
```
Comment (MohitGarg2905) [inline parking/src/parking/slot.java:6]: wrong name

**Review by MohitGarg2905 — CHANGES_REQUESTED** [review]

File: parking/src/models/ParkingLot.java (hunk 5 above)
Comment (Girrajgg) [reply parking/src/models/ParkingLot.java:53]: slot status tells whether the slot of that particular floor is occupied or not, 0 = not occupied.
floor status tells the overall occupancy of that particular floor.

**Pull request closed without merging**